File: ssqq.capacitor-onebot-connector/ios/Sources/OnebotPlugin/Onebot.swift
import Foundation
import Network
import os

@available(iOS 14.0, *)
@objc public class Onebot: NSObject {

    // 需要返回 type 和 data 两个字段
    var onEvent: ((String, String) -> Void)?

    private var client: WebSocketClient?

    @objc public func connect(_ url: String) {

        let url = URL(string: url)!
        if(client != nil) {
            client?.disconnect()
        }
        // 使用 GCD 创建 WebSocket 连接
        DispatchQueue.global().async {
            let client = WebSocketClient(url: url, onEvent: { type, data in
                if let onEvent = self.onEvent {
                    onEvent(type, data)
                }
            })
            client.connect()
            // 全局保存 client 对象，避免被销毁
            self.client = client
        }
    }

    @objc public func send(_ data: String) {
        client?.sendMessage(data)
    }

    @objc public func close() {
        client?.disconnect()
    }

    @objc public func findService() {
        DispatchQueue.global().async {
            Task {
                let lanScanner = LANScanner(onEvent: { type, data in
                if let onEvent = self.onEvent {
                    onEvent(type, data)
                }
            })
                await lanScanner.scanCurrentLAN()
            }
        }
    }
}

File: ssqq.capacitor-onebot-connector/android/src/main/java/cn/stapxs/qqweb/connector/Onebot.java
package cn.stapxs.qqweb.connector;

import android.content.Context;

public class Onebot {
    public WebSocketClient connect(OnebotPlugin plugin, WebSocketClient webSocketClient, String value) {
        WebSocketClient client = null;
        if(webSocketClient == null) {
            client = new WebSocketClient(plugin);
            client.connect(value);
        }
        return client;
    }

    public Boolean send(WebSocketClient webSocketClient, String value) {
        if(webSocketClient != null) {
            webSocketClient.sendMessage(value);
        }
        return true;
    }

    public Boolean close(WebSocketClient webSocketClient) {
        if(webSocketClient != null) {
            webSocketClient.close();
        }
        return true;
    }

    public Boolean findService(OnebotPlugin plugin) {
        ScanNetwork scanNetwork = new ScanNetwork(plugin.getBridge().getContext(), plugin);
        scanNetwork.scanNetwork();
        return true;
    }
}
